File: fc-large2/ticketing/c-spring-cache/src/main/java/com/example/cspringcache/domain/repository/UserRepository.java
package com.example.cspringcache.domain.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.example.cspringcache.domain.entity.User;

public interface UserRepository extends JpaRepository<User, Long> {

	Optional<User> findByName(String name);

}

File: inf-java-to-kotlin-springboot/src/main/kotlin/com/group/libraryapp/domain/user/UserRepository.kt
package com.group.libraryapp.domain.user

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface UserRepository: JpaRepository<User, Long> {

    fun findByName(name: String): User?

    @Query(value = "select distinct u from User u left join fetch u.userLoanHistories ")
    fun findAllWithHistories(): List<User>

}
